### ISSUE TITLE
BUG: poly1d.coeffs is now a copy, not the underlying data

### ISSUE THREAD

**Issue by eric-wieser** [numpy/numpy#9385]
As raised by @matthew-brett [on the mailing list](http://numpy-discussion.10968.n7.nabble.com/Polynomial-silent-breakage-with-1-13-td44526.html), this is a regression in 1.13

> 
> ```python
> import numpy as np
> 
> poly = np.poly1d([1])
> poly.c[0] *= 2
> print(poly.c)
> ```
> 
> Numpy 1.12 gives (to me) expected output:
> ```
> [2]
> ```
> Numpy 1.13 gives (to me) unexpected output:
> ```
> [1]
> ```
> 
> The problem is caused by the fact that the coefficients are now a
> *copy* of the actual coefficient array - I think in an attempt to stop
> us modifying the coefficients directly.
> 
> I can't see any deprecation warnings with `-W always`.
> 
> The pain point here is that code that used to give the right answer
> has now (I believe silently) switched to giving the wrong answer.
> 

This was introduced in #8788, and a test verifying the opposite of this example in #8807(where this problem was actually [preempted](https://github.com/numpy/numpy/pull/8807#discussion_r107287021)), in order to avoid this bizarreness in 1.12:

```
>>> poly.c *= 2
ValueError: cannot set attribute
>>> poly.c
<doubled anyway>
```

Perhaps instead, we should just allow `coeffs` to be modified in the first place, rather than breaking compatibility by fixing half-assed immutability

**Comment by eric-wieser** [comment]
Another option would be to define
```
@coeffs.setter
def coeffs(self, value):
    if value is not self._coeffs:
        raise AttributeError
```

Which makes `poly.c *= 2` legal, but leaves `poly.c = np.array(...)` illegal, like it always (arbitrarily) has been

**Comment by mhvk** [comment]
I think I'd prefer just making them mutable over adding yet another layer...  But maybe there is a more elegant solution.

**Comment by charris** [comment]
> I think I'd prefer just making them mutable

@mhvk That was my first thought also.

**Comment by matthew-brett** [comment]
I don't know that part of the code at all well, but that's my general preference too.

**Comment by eric-wieser** [comment]
This was fixed in #9390, but keeps `id(poly.c)` immutable, in case subclasses rely upon that.
